app-modules/meeting-center/src/GoogleCalendarManager.php

```diff
@@ -0,0 +1,205 @@
+         * the event invitation email.
+         *
+         * @see https://developers.google.com/calendar/api/v3/reference/events/insert
+         *
+         * @todo check for overlapping events
+         * @todo add attendees?
+         * @todo auto accept?
+         * @todo create later?
+         * */
+        $service = (new GoogleCalendar(static::client($event->calendar)));
+
+        $googleEvent = $this->toGoogleEvent($event);
+        $googleEvent = $service->events->insert($event->calendar->provider_id, $googleEvent);
+        $event->provider_id = $googleEvent->getId();
+
+        return $event;
+    }
+
+    public function updateEvent(CalendarEvent $event): CalendarEvent
+    {
+        if ($event->provider_id) {
+            $googleEvent = $this->toGoogleEvent($event);
+            $service = (new GoogleCalendar(static::client($event->calendar)));
+            $service->events->update($event->calendar->provider_id, $event->provider_id, $googleEvent);
+        } else {
+            $event = $this->createEvent($event);
+        }
+
+        return $event;
+    }
+
+    public function deleteEvent(CalendarEvent $event): void
+    {
+        $service = (new GoogleCalendar(static::client($event->calendar)));
+        $service->events->delete($event->calendar->provider_id, $event->provider_id);
+    }
+
+    public function syncEvents(Calendar $calendar): void
+    {
+        $events = collect($this->getEvents($calendar));
+
+        $events
+            ->each(
+                function (Event $event) use ($calendar) {
+                    $userEvent = $calendar->events()->where('provider_id', $event->id)->first();
+
+                    if ($userEvent) {
+                        $userEvent
+                            ->updateQuietly([
+                                'title' => $event->summary,
+                                'description' => $event->description,
+                                'starts_at' => $event->start->dateTime,
+                                'ends_at' => $event->end->dateTime,
+                            ]);
+                    } else {
+                        $calendar
+                            ->events()
+                            ->createQuietly([
+                                'provider_id' => $event->id,
+                                'title' => $event->summary,
+                                'description' => $event->description,
+                                'starts_at' => $event->start->dateTime,
+                                'ends_at' => $event->end->dateTime,
+                            ]);
+                    }
+                }
+            );
+
+        $calendar->events()
+            ->whereNull('provider_id')
+            ->each(fn ($event) => $this->createEvent($event)->saveQuietly());
+
+        $calendar->events()->whereNotIn('provider_id', $events->pluck('id'))->delete();
```

Comment:
Won't this delete the events that happened yesterday and beyond out of our system?

We don't have to solve for this now, but we might want to add a TODO here to look into this issue and determine how/what event info we should be retaining in our system.